napari/layers/base/base.py

```diff
@@ -181,11 +181,14 @@ def __init__(
         self.dims.events.order.connect(self._update_dims)
         self.dims.events.axis.connect(self.refresh)
 
-        self.mouse_move_callbacks = []
-        self.mouse_drag_callbacks = []
+        # self.mouse_move_callbacks = []
+        # self.mouse_drag_callbacks = []
         self._persisted_mouse_event = {}
         self._mouse_drag_gen = {}
 
+    mouse_move_callbacks = []  # Note this is now at class level
```

Comment:
Here is the problem code that I need help fixing

Reply:
If you want the behavior we discussion in person (i.e., decorator to set up the default callbacks but with exceptions on a per-layer basis), I think you're stuck with this strategy:

1. Create class callbacks like what you have.
2. Create a list of instance additions to the class callbacks.
3. Create a list of instance removals from the class callbacks.

Yeah, it's ugly.  :(

Reply:
@sofroniewn I think we should hold off on class vs instance callbacks for now. How I did it in [my branch](https://github.com/kne42/napari/tree/refactor-internal-mouse-bindings) was by keeping the callbacks list at the instance level and instead of using decorators to register them as defaults, inside the `mode` setter you choose which one you want.

Example: https://github.com/kne42/napari/blob/339b1271aa8591913ebdb035293931a05c2ce5e1/napari/layers/points/points.py#L1094-L1118

Reply:
Thanks for the guidance @ttung and the example of how you tried it earlier @kne42. As you mention, in your approach you put the binding inside the mode setter. I was trying to pull the binding out into its own file, but that is proving tricky. 

@shanaxel42 what do you make of this / how do you recommend we proceed given the refactor planned in #1020. The goal of this series of PRs would have been to pull the mouse stuff out a little, but maybe it's not worth it. It could be very useful for #1020 to think about what adding an example of with a mouse function / keybindings callback (maybe from the Labels layer) would look like and then determine if this refactor helps or not, or what it should look like.

Reply:
I have now taken the approach that @kne42 implemented in her branch. This PR is now almost identical to that one, but just for the labels layer, and with some additional naming changes.